pkg/executor/executor.go

```diff
@@ -0,0 +1,49 @@
+package executor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type Interface interface {
+	Execute(ctx context.Context, argsf string, a ...interface{}) ([]byte, error)
+}
+
+type executor struct {
+	cmd     string
+	prepend []string
+}
+
+var _ Interface = &executor{}
+
+func NewExecutor(cmd string, prepend ...string) Interface {
+	return &executor{
+		cmd:     cmd,
+		prepend: prepend,
+	}
+}
+
+// Execute executes the command with the given arguments.
+func (e *executor) Execute(ctx context.Context, argsf string, a ...interface{}) ([]byte, error) {
+	substituted := fmt.Sprintf(argsf, a...)
+	args := append(e.prepend, strings.Split(substituted, " ")...)
```

Comment:
>args := append(e.prepend, strings.Split(substituted, " ")...) [](http://example.com/codeflow?start=1&length=61)

What if `a` contains spaces?

For instance, `"%s %s", "foo", "foo bar"` will become `["foo", "foo", "bar"]`, not `[ "foo", "foo bar"]`.